container_opts.go

```diff
@@ -165,9 +166,18 @@ func WithNewSnapshotView(id string, i Image) NewContainerOpts {
 	}
 }
 
-func setSnapshotterIfEmpty(c *containers.Container) {
+func setSnapshotterIfEmpty(ctx context.Context, client *Client, c *containers.Container) {
 	if c.Snapshotter == "" {
-		c.Snapshotter = DefaultSnapshotter
+		defaultSnapshotter := DefaultSnapshotter
+		namespaceService := client.NamespaceService()
+		if ns, err := namespaces.NamespaceRequired(ctx); err == nil {
+			if labels, err := namespaceService.Labels(ctx, ns); err == nil {
+				if snapshotLabel, ok := labels["containerd.io/defaults/snapshotter"]; ok {
```

Comment:
how about the const for the `label` here?

Reply:
You mean to use defaults package for the label names? That sounds better, should I move both label names to https://github.com/containerd/containerd/blob/master/defaults/defaults.go?

Reply:
it is ok to move it to defaults